com.unity.cinemachine/Editor/Samples/SampleDependencyImporter.cs

```diff
@@ -76,53 +80,66 @@ void LoadAssetDependencies(string assetPath)
                             
                             // Import sample-specific asset dependencies
-                            assetsImported |=
-                                ImportAssetDependencies(m_PackageInfo, sampleEntry.AssetDependencies);
+                            localAssetsImported = ImportAssetDependencies(
+                                m_PackageInfo, sampleEntry.AssetDependencies, 
+                                out bool localAssetsReimported);
                             
                             // Import common amd sample specific package dependencies
                             m_PackageDependencyIndex = 0;
                             m_PackageDependencies = new List<string>(m_SampleConfiguration.SharedPackageDependencies);
                             m_PackageDependencies.AddRange(sampleEntry.PackageDependencies);
-                            if (m_PackageDependencies.Count != 0 && PromptUserConfirmation(m_PackageDependencies))
+                            
+                            if (m_PackageDependencies.Count != 0 && 
+                                DoDependenciesNeedToBeImported(out var dependenciesToImport))
                             {
-                                // Import package dependencies using the editor update loop, because
-                                // adding packages need to be done in sequence one after the other
-                                EditorApplication.update += ImportPackageDependencies;
+                                switch (PromptUserConfirmation(dependenciesToImport))
+                                {
+                                    case UserChoice.ImportPackageDependencies:
+                                        // only import dependencies that are missing
+                                        m_PackageDependencies = dependenciesToImport;
+                                        // Import package dependencies using the editor update loop, because
+                                        // adding packages need to be done in sequence one after the other
+                                        EditorApplication.update += ImportPackageDependencies;
+                                        break;
+                                    case UserChoice.AbortImport:
+                                        // if samples were already imported, don't delete them
+                                        if (!localAssetsReimported)
+                                            DeleteSampleImported(sample.importPath, 
+                                                !sharedAssetsReimported, m_SampleConfiguration.SharedAssetDependencies,
+                                                localAssetsImported, sampleEntry.AssetDependencies);
+                                        break;
+                                    default:
+                                    case UserChoice.ImportSampleButNotPackageDependencies:
```

Comment:
If the user presses the little x at the top-right of the dialog, what happens?  The default action?  It should be Abort in that case.

Reply:
That X is a windows only thing. I don't have that O.O

But according to the docs, X will do the second input option (returning 1) - > ImportSampleButNotPackageDependencies

So I would need to change the order so that AbortImport is 1.

Reply:
I'm now of the opinion that we should remove the "Abort" option, since it's kind of a hack and technically it's too late to stop it